datafusion/physical-expr/src/string_expressions.rs

```diff
@@ -296,6 +296,50 @@ pub fn initcap<T: OffsetSizeTrait>(args: &[ArrayRef]) -> Result<ArrayRef> {
     Ok(Arc::new(result) as ArrayRef)
 }
 
+/// Returns the position of the first occurrence of substring in string.
+/// The position is counted from 1. If the substring is not found, returns 0.
+/// For example, instr('Helloworld', 'world') = 6.
+pub fn instr<T: OffsetSizeTrait>(args: &[ArrayRef]) -> Result<ArrayRef> {
```

Comment:
I hunted around in the arrow kernels and I did not find an equivalent

@tustvold  do you know off hand if there is an arrow compute kernel that compute the equivalent of `instr`?